app/main.py

```diff
@@ -1,2 +1,102 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any, next_node: Any = None) -> None:
```

Comment:
This class better be defined inside of a Dictionary since it is a part of its implementation and will not be used outside of it. Also use correct type annotations.
I see that you are implementing a link list to avoid collisions but this will not be the most effective approach in terms of performance - insertion will be fast but retrieval of data will be O(n) and we are aiming for O(1). Try to follow the example in the case study